### ISSUE TITLE
Question about dataset filtering in Section 3.5: Cannot reproduce ~10M samples

### ISSUE THREAD

**Issue by undefined443** [M-E-AGI-Lab/Muddit#9]
Hi, thank you for the great work!

I'm trying to reproduce the data filtering pipeline described in **Section 3.5 (The Scalability of Muddit)** of the **ICLR 2026 submission version** of the paper, but I found significant discrepancies between the paper's claimed dataset size and my filtering results.

### Paper Description (Section 3.5, Lines 471-476)

> "To demonstrate the scalability of our approach, we curate roughly **10 million** image–text pairs from LAION-ART, JourneyDB, and CC12M. We filter out samples with an aesthetic score below 7, a height or width under 512 pixels, or an aspect ratio above 2."

### My Filtering Method

I used the [Improved Aesthetic Predictor](https://github.com/christophschuhmann/improved-aesthetic-predictor) with the following setup:

- **CLIP Model**: `ViT-L/14`
- **Aesthetic Predictor Weights**: [`sac+logos+ava1-l14-linearMSE.pth`](https://github.com/christophschuhmann/improved-aesthetic-predictor/raw/main/sac%2Blogos%2Bava1-l14-linearMSE.pth)
- **Thresholds**: `aesthetic >= 7`, `min(width, height) >= 512`, `aspect_ratio <= 2`

<details>
<summary>Aesthetic Scoring Code</summary>

```python
import clip
import torch

class AestheticMLP(torch.nn.Module):
    def __init__(self, input_size=768):
        super().__init__()
        self.layers = torch.nn.Sequential(
            torch.nn.Linear(input_size, 1024),
            torch.nn.Dropout(0.2),
            torch.nn.Linear(1024, 128),
            torch.nn.Dropout(0.1),
            torch.nn.Linear(128, 64),
            torch.nn.Dropout(0.1),
            torch.nn.Linear(64, 16),
            torch.nn.Linear(16, 1),
        )

    def forward(self, x):
        return self.layers(x)

# Load models
clip_model, preprocess = clip.load("ViT-L/14", device="cuda")
aesthetic_model = AestheticMLP(input_size=768)
aesthetic_model.load_state_dict(torch.load("sac+logos+ava1-l14-linearMSE.pth"))
aesthetic_model.to("cuda").eval()

# Calculate score
with torch.no_grad():
    image_tensor = preprocess(image).unsqueeze(0).to("cuda")
    features = clip_model.encode_image(image_tensor)
    features = features / features.norm(dim=-1, keepdim=True)
    score = aesthetic_model(features.float()).item()
```

</details>

For LAION-Art, I used the pre-computed `aesthetic` field in the dataset metadata.

#### 1. LAION-Art (8.06M images)

Using the pre-computed aesthetic scores from the dataset:

| Filter | Pass Count | Pass Rate |
|--------|-----------|-----------|
| Aesthetic >= 7 | 8,062,077 | 100% (already >= 8) |
| Resolution >= 512 | 2,376,769 | 29.48% |
| Aspect Ratio <= 2 | 7,932,410 | 98.39% |
| **All Conditions** | **2,355,765** | **29.22%** |

#### 2. CC12M

| Metric | Value |
|--------|-------|
| After resolution & aspect ratio filter | 6,800,594 |
| Aesthetic >= 7 | **599** (0.01%) |
| Aesthetic >= 6 | 184,625 (2.71%) |
| Mean aesthetic score | 5.04 |

#### 3. JourneyDB (4.19M images)

| Metric | Value |
|--------|-------|
| After resolution & aspect ratio filter | 4,185,589 |
| Aesthetic >= 7 | **379,272** (9.1%) |
| Aesthetic >= 6.5 | 1,610,639 (38.5%) |
| Aesthetic >= 6 | 3,093,789 (73.9%) |
| Mean aesthetic score | 6.31 |

### Summary

| Dataset | Pass All Filters |
|---------|-----------------|
| LAION-Art | ~2.36M |
| JourneyDB | **379,272** |
| CC12M | ~600 |
| **Total** | **~2.74M** |

This is approximately **3.6x less** than the ~10M claimed in the paper.

### Questions

1. Is there a different aesthetic scoring model or weights used?
2. Is the resolution threshold different (e.g., shorter side vs. both sides)?
3. Are there additional data sources not mentioned in the paper?
4. Could you share the exact filtering script for reproducibility?

Thank you for your clarification!

**Comment by Shi-qingyu** [comment]
Hi! Thank you for your interest in our work. Your conclusion is correct. Setting an aesthetic score of 7 as the threshold will cover most of the data in CC12M and JourneyDB. Our 10M dataset includes a 6M internal dataset, and we will update the description in the paper. Additionally, to expand the dataset, the aesthetic score can be lowered to 6 or 6.5.

**Comment by viiika** [comment]
To better replace our 6M internal high-quality dataset, we recommend using state-of-the-art text-to-image models currently available, such as nano-banana, z-image, and flux, to construct image-text pairs (this is often much faster than downloading images via URLs). Additionally, you are welcome to use our dataset ([https://huggingface.co/MeissonFlow](https://huggingface.co/MeissonFlow)) or BLIP3o’s dataset ([https://huggingface.co/BLIP3o](https://huggingface.co/BLIP3o)), such as [https://huggingface.co/datasets/BLIP3o/BLIP3o-Pretrain-Long-Caption](https://huggingface.co/datasets/BLIP3o/BLIP3o-Pretrain-Long-Caption).
